src/index.js

```diff
@@ -1,22 +1,22 @@
 // @flow
-type EqualityFn = (a: any, b: any) => boolean;
+type EqualityFn = (a: mixed, b: mixed) => boolean;
 
-const simpleIsEqual = (a: any, b: any): boolean => a === b;
+const simpleIsEqual: EqualityFn = (a: mixed, b: mixed): boolean => a === b;
 
-export default function (resultFn: Function, isEqual?: EqualityFn = simpleIsEqual) {
-  let lastThis: any;
-  let lastArgs: Array<any> = [];
-  let lastResult: any;
+export default function <ResultFn: Function>(resultFn: ResultFn, isEqual?: EqualityFn = simpleIsEqual): ResultFn {
```

Comment:
@thejameskyle I was unable to get a better check than:

```js
<ResultFn: Function>
```

I tried to use this:

```js
<ResultFn: () => mixed>
```

but flow complained when I tried to memoize functions that had arguments. I also tried this with no success.

```js
<ResultFn: (...args: Array<mixed>)>
```